avm/res/network/network-watcher/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,149 @@
+targetScope = 'subscription'
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'NetworkWatcherRG' // Note, this is the default NetworkWatcher resource group. Do not change.
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nnwwaf'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+var resourceGroupLocation = resourceLocation == 'westeurope' ? 'westeurope' : 'westeurope'
```

Comment:
Not sure I fully grasp this condition. If we are assigning the same value in any case, why not directly assigning the location?

Reply:
@ChrisSidebotham wants to have `resourceLocation` not sitting there as unused param.

Reply:
I'd argue adding the corresponding suppression with a comment would raise less eyebrows 😄

Reply:
final changes done